accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,221 @@
+# npm workspaces: Running commands
+
+## Summary
+
+Introduces basic support to running **npm commands** across nested **workspaces**.
+
+## Motivation
+
+The need for running **npm commands** across nested **workspaces** was sourced from the community during the elaboration of the [original **npm workspaces** RFC](https://github.com/npm/rfcs/pull/103).
+
+## Detailed Explanation
+
+Following up from the original **npm workspaces** RFC are the following command-related changes:
+
+- A subset of the standard **npm commands** needs to be made aware of **npm workspaces**
+- It's desired to have a standard way to filter out **workspaces** in which to run specific **npm commands**
+
+## Rationale and Alternatives
+
+The ability to run **npm commands** across defined **workspaces** is essential to successfully manage a **npm workspaces** workflow. That makes the following alternatives much less desireable:
+
+- Do not implement further support to manage running commands across **workspaces**
+- Implement only _some_ of the implementation changes described in this document
+
+**Note:** Alternative ways of executing each of the described features are defined along with their implementation details below.
+
+## Implementation
+
+### 1. Run commands across all child packages
+
+Make **npm cli** subcommands **npm workspaces**-aware, so that running a command tries to run it within all **workspaces** as long as a **workspaces configuration** field is properly defined in `package.json`.
+
+Only a subset of commands are to be supported:
+
+- `fund` List funding info for all **workspaces**
+- `ls` List all packages including **workspaces**
+- `outdated` List outdated **dependencies** including **workspaces** and its **dependencies**
+- `run-script` Run arbitrary **scripts** in all **workspaces**, skip any **workspace** that does not have a targetting **script**
```

Comment:
`npm exec` would be another awesome addition for workspaces. Oftentimes workspace monorepos need to write scripts that run on every package. Needing to do stuff like this is super common:

```sh
npm exec -- $(pwd)/scripts/update-readme-toc.sh
```

This would evaluate the script `/path/to/project/scripts/update-readme-toc.sh` on every workspace directory as a separate process with `/workspace/node_modules/.bin:/node_modules/.bin` in the `PATH`

Reply:
this was brought up once again during today's OpenRFC call 😄 I'll make sure to add it to the RFC but it will probably live in `npm workspaces exec` (`npm ws exec`) since in **npm7** we already have a `npm exec` top-level command.